pkgs/development/tools/misc/libtool/libtool2.nix

```
@@ -36,6 +37,29 @@ stdenv.mkDerivation rec {
     popd
   '';
 
+  # When detecting the `LD` (linker) command and flags for cross
+  # compiling, libtool.m4 expects to be able to use "/usr/bin/file" to
+  # detect host details like ABI and linker flags.  Unfortunately this
+  # detection code hardwires the path to /usr/bin/file, which is
+  # inaccessible for sandboxed builds.  This causes the `LD` flag to
+  # be detected incorrectly when cross-compiling to the following
+  # hostPlatforms: {x86_64, powerpc, s390, sparc, mips64*}-linux,
+  # x86_64-kfreebsd, *-solaris, *-irix, and *-hpux.
+  #
+  # This substitution is performed only for cross-compilation in order
+  # to avoid a mass-rebuild.  It can be removed when nixpkgs upgrades
+  # to libtool 2.4.7, which fixes this problem:
+  #
+  #   https://git.savannah.gnu.org/cgit/libtool.git/commit/?h=branch-1-5&id=da2e352735722917bf0786284411262195a6a3f6
+  #
+  preConfigure = if stdenv.buildPlatform == stdenv.targetPlatform
+                 then null
+                 else ''
+                 substituteInPlace m4/libtool.m4     --replace "/usr/bin/file" "${buildPackages.file}/bin/file"
+                 substituteInPlace configure         --replace "/usr/bin/file" "${buildPackages.file}/bin/file"
+                 substituteInPlace libltdl/configure --replace "/usr/bin/file" "${buildPackages.file}/bin/file"
+                 '';
```

```suggestion
  preConfigure = lib.optionalString (stdenv.buildPlatform != stdenv.targetPlatform) ''
    for file in configure libltdl/configure m4/libtool.m4; do
      substituteInPlace $file  --replace "/usr/bin/file" "${buildPackages.file}/bin/file"
    done
  '';
```

that should be equivalent but shorter and sleaker.